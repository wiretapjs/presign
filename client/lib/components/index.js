export { default as Button } from './Button/Button'
export { default as Background } from './Background/Background'
export { default as Card } from './Card/Card'
export { default as ErrorMessage } from './ErrorMessage/ErrorMessage'
export { default as Paper } from './Paper/Paper'
export { default as EnvironmentWatermark } from './EnvironmentWatermark/EnvironmentWatermark'
export { default as Header } from './Header/Header'
export { default as LegalText } from './LegalText/LegalText'
export { default as Logo } from './Logo/Logo'
export { default as MainContainer } from './MainContainer/MainContainer'
export { default as MainTitle } from './MainTitle/MainTitle'
export { default as Column } from './Column/Column'
export { default as Row } from './Row/Row'
export { default as Span } from './Span/Span'
export { default as Footer } from './Footer/Footer'
export { default as EsignIframe } from './Iframe/EsignIframe'
export { default as HelpMessage } from './HelpMessage/HelpMessage'
