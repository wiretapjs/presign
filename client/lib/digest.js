export default function (string) {
  return string.split('').reduce((accumulator, character) => {
    accumulator = ((accumulator << 5) - accumulator) + character.charCodeAt(0)
    return accumulator & accumulator
  }, 0)
}
