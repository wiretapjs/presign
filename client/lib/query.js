import QueryString from 'query-string'

const get = function (param) {
  return QueryString.parse(window.location.search)[param]
}

export default { get }
