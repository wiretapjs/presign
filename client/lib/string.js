export const capitalize = function (string) {
  if (!string) throw new Error('Undefined input')
  return string.charAt(0).toUpperCase() + string.slice(1)
}
