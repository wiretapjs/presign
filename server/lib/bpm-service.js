const request = require('request')
const HttpStatus = require('http-status-codes')
const _ = require('lodash')
const Url = require('url-parse')
const RemoteServiceError = require('models/errors/remoteService')
const config = require('../config')

const DEFAULT_REQUEST_OPTIONS = {
  json: true,
  proxy: config.proxy,
}

const isErrorResponse = function (err, response) {
  return err || response.statusCode >= HttpStatus.BAD_REQUEST || _.get(response, 'body.error')
}

const formatRequest = function (options) {
  const requestConfig = Object.assign({}, DEFAULT_REQUEST_OPTIONS, options)
  requestConfig.headers.host = getRequestHost(requestConfig.url)

  return requestConfig
}

const invoke = function (requestConfig) {
  requestConfig = formatRequest(requestConfig)
  return new Promise((resolve, reject) => {
    request(requestConfig, (err, response) => {
      if (isErrorResponse(err, response)) {
        return reject(new RemoteServiceError(`Failed to get ${requestConfig.url}`, {
          response,
          internalError: err,
          headers: _.get(response, 'headers'),
        }))
      }
      return resolve(response)
    })
  })
}

const get = function (requestConfig) {
  requestConfig.method = 'GET'
  return invoke(requestConfig)
}

const getRequestHost = function (requestUrl) {
  return new Url(requestUrl).host
}

module.exports = { get, invoke }
