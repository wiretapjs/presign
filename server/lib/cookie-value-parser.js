class CookieValueParser {
  filterCookies (cookies, filter) {
    if (!cookies || !filter) {
      return []
    }

    const filteredCookies = cookies.filter(cookie => cookie.toLowerCase()
      .includes(filter.toLowerCase()))

    return filteredCookies
  }
}

module.exports = new CookieValueParser()
