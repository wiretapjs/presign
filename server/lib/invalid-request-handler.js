const ResultEnvelope = require('models/resultEnvelope')
const InvalidRequestError = require('models/errors/invalidRequest')

const handleInvalidRequest = function (validationResponse) {
  return Promise.reject(ResultEnvelope.Error(
    new InvalidRequestError(
      `Validation Errors occured when validating your request.`,
      {
        validationErrors: validationResponse.errors,
      })))
}

module.exports = handleInvalidRequest
