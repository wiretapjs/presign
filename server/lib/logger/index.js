const pino = require('pino')()

const sendMessage = message => {
  const { type = 'info', text = '' } = message
  if (text) {
    if (type === 'error') pino.error(text)
    pino.info(text)
  }
  return message
}

module.exports = {
  sendMessage,
}
