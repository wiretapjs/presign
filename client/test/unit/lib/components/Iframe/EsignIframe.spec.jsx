import React from 'react'
import { shallow, mount } from 'enzyme'
import { EsignIframe } from 'lib/components'

describe('EsignIframe', () => {
  it('should render correctly', () => {
    const esignIframe = shallow(<EsignIframe />)
    esignIframe.setState({showLoading: true})
    esignIframe.iframeOnLoadEvent = jest.fn()
    expect(esignIframe).toMatchSnapshot()
  })
  it('should set show loading to false once iframe loaded', () => {
    const esignIframe = mount(<EsignIframe url="test" esignEvents={jest.fn()} />)
    esignIframe.receiveMessage = jest.fn()
    const iFrame = esignIframe.find('iframe')
    expect(iFrame.exists()).toBe(true)
    iFrame.simulate('load')
    expect(esignIframe.state().showLoading).toBe(false)
  })
})

describe('After mounting', () => {
  let receiveMessage, mountEsignInstance
  const mountEsignIframe = () => {
    const mountEsignWrapper = shallow(<EsignIframe url="test" esignEvents={jest.fn()} />)
    mountEsignInstance = mountEsignWrapper.instance()
    receiveMessage = jest.spyOn(mountEsignInstance, 'receiveMessage')
    mountEsignInstance.componentDidMount()
    window.dispatchEvent(new CustomEvent('message'))
    return mountEsignInstance
  }
  it('should recive message from iframe', async () => {
    await mountEsignIframe()
    expect(receiveMessage).toHaveBeenCalled()
  })
  it('should post ESL:MESSAGE:REGISTER', async () => {
    const event = {
      origin: 'libertymutual.com',
      data: 'ESL:MESSAGE:REGISTER',
      source: {
        postMessage: jest.fn(),
      },
    }
    mountEsignInstance.receiveMessage(event)
    expect(event.source.postMessage).toHaveBeenCalledWith('ESL:MESSAGE:ACTIVATE_EVENTS', 'libertymutual.com')
  })
  it('should post ESL:MESSAGE:SUCCESS:SIGNER_COMPLETE', async () => {
    const event = {
      origin: 'libertymutual.com',
      data: 'ESL:MESSAGE:SUCCESS:SIGNER_COMPLETE',
      source: {
        postMessage: jest.fn(),
      },
    }
    mountEsignInstance.receiveMessage(event)
    expect(event.source.postMessage).toHaveBeenCalledWith('ESL:MESSAGE:SUCCESS:SIGNER_COMPLETE', 'libertymutual.com')
  })
  it('should post any other events back to iframe', async () => {
    const event = {
      origin: 'libertymutual.com',
      data: 'events',
      source: {
        postMessage: jest.fn(),
      },
    }
    mountEsignInstance.receiveMessage(event)
    expect(event.source.postMessage).toHaveBeenCalledWith('events', 'libertymutual.com')
  })
})
