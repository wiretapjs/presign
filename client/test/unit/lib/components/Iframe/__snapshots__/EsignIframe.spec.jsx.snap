// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EsignIframe should render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "iframeOnLoadEvent": [Function],
  "length": 1,
  "node": <styled.div>
    <styled.div>
       Loading... 
    </styled.div>
    <styled.iframe
      frameBorder="0"
      onLoad={[Function]}
      src={undefined}
    />
  </styled.div>,
  "nodes": Array [
    <styled.div>
      <styled.div>
         Loading... 
      </styled.div>
      <styled.iframe
        frameBorder="0"
        onLoad={[Function]}
        src={undefined}
      />
    </styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <EsignIframe />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": EsignIframe {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "showLoading": true,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div>
          <styled.div>
             Loading... 
          </styled.div>
          <styled.iframe
            frameBorder="0"
            onLoad={[Function]}
            src={undefined}
          />
        </styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div>
          <styled.div>
             Loading... 
          </styled.div>
          <styled.iframe
            frameBorder="0"
            onLoad={[Function]}
            src={undefined}
          />
        </styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <EsignIframe />,
}
`;
